### PR TITLE
isort select submodules

### PR DESCRIPTION
**Pull request by nstarman**
Astropy's existing isort options are moved to pyproject.toml and integrated with pre-commit.
Almost everything in astropy is ignored by default except for specific submodules.
To allow a submodule to have consistent, sorted imports, the relevant ignore glob may be removed.

@WilliamJamieson. Feel free to un-skip modeling and add a commit to this PR!

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/cosmology/tests/test_parameters.py [hunk 1]
```diff
@@ -3,9 +3,10 @@
 # STDLIB
 from types import MappingProxyType
 
+import pytest
```
Comment (pllim) [inline astropy/cosmology/tests/test_parameters.py:6]: This is a step too far? LoL. Isn't `pytest` technically third-party like `numpy`?

File: astropy/cosmology/tests/test_funcs.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
 from astropy import units as u
 from astropy.cosmology import core, flrw
 from astropy.cosmology.funcs import _z_at_scalar_value, z_at_value
-from astropy.cosmology.realizations import (WMAP1, WMAP3, WMAP5, WMAP7, WMAP9,
-                                            Planck13, Planck15, Planck18)
+from astropy.cosmology.realizations import (WMAP1, WMAP3, WMAP5, WMAP7, WMAP9, Planck13, Planck15,
```
Comment (pllim) [inline astropy/cosmology/tests/test_funcs.py:14]: Wow, it is even nitpickier than me. Why so picky on the line length?

File: astropy/cosmology/tests/test_parameters.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/tests/test_parameters.py:6]: Lol. yes. what's happening here is numpy has it's own section (I'm not sure why astropy had this historically) in the isort config, so pytest is getting moved up, but the section header is not. I'll just tweak that. 

I'd also rather not separate out numpy in the config...

Comment (nstarman) [reply astropy/cosmology/tests/test_parameters.py]: Pushing a new commit consolidating import headings

Comment (pllim) [reply astropy/cosmology/tests/test_parameters.py]: Let's just keep the basic sorting part of isort and not try to make it too fancy? 😬

Comment (nstarman) [reply astropy/cosmology/tests/test_parameters.py]: Done. We can always re-implement the auto-labelling.

**Review by pllim — APPROVED** [review]
I think this is a good start. Thanks!

**Comment by pllim** [comment]
Is this good to go, or are you waiting for others, @nstarman ?

**Comment by nstarman** [comment]
Either or. @WilliamJamieson and I were discussing that this should be implemented and that isort should be applied to modeling. But there's no reason that has to happen in this PR.

**Comment by pllim** [comment]
modeling can be a follow-up PR then. Thanks!

**Pull request merged**